app/src/main/java/org/schabi/newpipe/util/MobileDataHelper.java

```diff
@@ -0,0 +1,44 @@
+package org.schabi.newpipe.util;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkCapabilities;
+import android.net.NetworkInfo;
+import android.os.Build;
+import android.preference.PreferenceManager;
+
+import org.schabi.newpipe.R;
+import org.schabi.newpipe.extractor.stream.StreamInfo;
+import org.schabi.newpipe.extractor.stream.StreamType;
+
+public class MobileDataHelper {
```

Comment:
We are already checking mobile data for selecting the stream based on how you connect. So a helper would confuse things up as the already existing solution is not using a helper. As this class is only used once on livestreams I suggest to you to also
- either name it `MobileDataLiveStreamHelper`
- or make the stream select mechanism use your solution as well.

Reply:
I renamed it